### PR TITLE
[Sparse] polish strings of matmul.py and sddmm.py.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/sparse/sddmm.py [hunk 1]
```diff
@@ -58,32 +58,33 @@ def sddmm(A: SparseMatrix, X1: torch.Tensor, X2: torch.Tensor) -> SparseMatrix:
...
-    for bsddmm.
+    The batch dimension is the last dimension for input dense matrices. In
+    particular, if the sparse matrix has scalar non-zero values, it will be
+    broadcasted for bsddmm.
 
     Parameters
     ----------
     A : SparseMatrix
-        Sparse matrix of shape ``(M, N)`` or ``(M, N, B)``.
+        Sparse matrix of shape ``(L, N)`` with scalar values or vector values of
```
Comment (czkkkkkk) [inline python/dgl/sparse/sddmm.py:77]:
```suggestion
        Sparse matrix of shape ``(L, N)`` with vector values of length ``K``.
```

File: python/dgl/sparse/sddmm.py [hunk 2]
```diff
@@ -58,32 +58,33 @@ def sddmm(A: SparseMatrix, X1: torch.Tensor, X2: torch.Tensor) -> SparseMatrix:
...
+        Dense matrix of shape ``(L, M, K)``
     X2 : Tensor
-        Dense matrix of shape ``(K, N, B)``
+        Dense matrix of shape ``(M, N, K)``
 
     Returns
     -------
     SparseMatrix
-        Sparse matrix of shape ``(M, N, B)``.
+        Sparse matrix of shape ``(L, N, K)``.
```
Comment (czkkkkkk) [inline python/dgl/sparse/sddmm.py:87]:
```suggestion
        Sparse matrix of shape ``(L, N)`` with vector values of length ``K``.
```

File: python/dgl/sparse/matmul.py [hunk 3]
```diff
@@ -142,24 +142,24 @@ def _sparse_diag_mm(A, D):
...
+    """Internal function for multiplying a diagonal matrix by a sparse matrix.
 
     Parameters
     ----------
     D : DiagMatrix
-        Matrix of shape (N, M), with values of shape (nnz1)
+        Diagonal matrix of shape ``(L, M)``
     A : DiagMatrix
-        Matrix of shape (M, P), with values of shape (nnz2)
+        Diagonal matrix of shape ``(M, N)``
```
Comment (czkkkkkk) [inline python/dgl/sparse/matmul.py:157]:
```suggestion
        Sparse matrix of shape ``(M, N)``
```

File: python/dgl/sparse/matmul.py [hunk 4]
```diff
@@ -124,14 +124,14 @@ def _sparse_diag_mm(A, D):
     Parameters
     ----------
     A : SparseMatrix
-        Matrix of shape (N, M), with values of shape (nnz1)
+        Sparse matrix of shape ``(L, M)``
     D : DiagMatrix
-        Matrix of shape (M, P), with values of shape (nnz2)
+        Sparse matrix of shape ``(M, N)``
```
Comment (czkkkkkk) [inline python/dgl/sparse/matmul.py:129]:
```suggestion
        Diagonal matrix of shape ``(M, N)``
```

**Review by czkkkkkk — APPROVED** [review]

File: python/dgl/sparse/matmul.py [hunk 5]
```diff
@@ -17,14 +17,14 @@ def spmm(A: Union[SparseMatrix, DiagMatrix], X: torch.Tensor) -> torch.Tensor:
     Parameters
     ----------
     A : SparseMatrix or DiagMatrix
-        Sparse matrix of shape (N, M) with values of shape (nnz)
+        Sparse matrix of shape ``(L, M)`` with scalar values
     X : torch.Tensor
-        Dense tensor of shape (M, F) or (M)
+        Dense matrix of shape ``(M, N)`` or ``(M)``
```
Comment (jermainewang) [inline python/dgl/sparse/matmul.py:22]: Is there any convention on why picking these alphabets?

Comment (frozenbugs) [reply python/dgl/sparse/matmul.py:22]: Not I am aware of any global convention.
I changed this to keep everything consistent in DGL sparse tutorial and doc.

File: python/dgl/sparse/sddmm.py (hunk 1 above)
Comment (frozenbugs) [reply python/dgl/sparse/sddmm.py:77]: I think we support broadcast, which allows scalar values in sparse matrix as well?

**Pull request merged**